packages/babel-plugin-transform-es2015-block-scoping/src/index.js

```diff
@@ -416,56 +416,79 @@ class BlockScoping {
     // build the closure that we're going to wrap the block with, possible wrapping switch(){}
     const fn = t.functionExpression(null, params,
       t.blockStatement(isSwitch ? [block] : block.body));
-    fn.shadow = true;
 
     // continuation
     this.addContinuations(fn);
 
-    let ref = fn;
-
-    if (this.loop) {
-      ref = this.scope.generateUidIdentifier("loop");
-      this.loopPath.insertBefore(t.variableDeclaration("var", [
-        t.variableDeclarator(ref, fn),
-      ]));
-    }
+    const has = this.has;
```

Comment:
I think this was leftover from copy-pasting from https://github.com/babel/babel/pull/5677/files/c8ff06f90661f5e67a93b7e2f2d387a029edd98a#diff-775ae6452de339480d54604c7d217b5dL463

Looks like `this.has` is used one other place, but otherwise isn't updated, so you're right I could move this up farther.